ccips/ccip-020/ccip-020-graceful-protocol-shutdown.md

```diff
@@ -0,0 +1,108 @@
+# CCIP-020
+
+## Preamble
+
+| CCIP Number   | 014                                   |
+| ------------- | ------------------------------------- |
+| Title         | Graceful Protocol Shutdown            |
+| Author(s)     | Jason Schrader jason@joinfreehold.com |
+| Consideration | Governance, Economic                  |
+| Type          | Standard                              |
+| Status        | Draft                                 |
+| Created       | 2024-03-22                            |
+| License       | BSD-2-Clause                          |
+| Supplements   | CCIP-012, CCIP-013, CCIP-017          |
+
+## Introduction
+
+As the CityCoins Protocol prepares for the upcoming Nakamoto release on the Stacks blockchain, the accumulation of technical debt with each change has made it increasingly challenging for core contributors and volunteers to keep up with the necessary updates.
+
+In light of these challenges and after careful consideration of community feedback, this CCIP proposes a graceful shutdown of the CityCoins Protocol. The shutdown process will involve disabling the mining and stacking functionality while ensuring that MIA and NYC token holders are fairly compensated through a redemption mechanism utilizing the STX held in the city treasuries.
+
+The primary objectives of this CCIP are to:
+
+- Protect the interests of MIA and NYC token holders
+- Honor the milestones achieved by the CityCoins Protocol
+- Honor the gift agreement with the City of Miami
+- Facilitate a smooth transition and minimize disruption to the ecosystem
+
+## Specification
+
+### Disable Mining
+
+This CCIP will disable mining in the current mining contract, such that:
+
+- Mining will stop at the block height the shutdown occurs
+- Any mining transactions sent after the shutdown block height will fail
+- Any mining claims made in this and prior contracts will still function
```

Comment:
We should set a deadline here if the DAO should be shut down totally.

Reply:
Deadline for which part? Mining and stacking claims have always been accessible to a user in all the contracts, and I don't think that should change?